mavutil.py

```diff
@@ -1036,15 +1039,25 @@ def recv(self,n=None):
             if e.errno in [ errno.EAGAIN, errno.EWOULDBLOCK, errno.ECONNREFUSED ]:
                 return ""
             raise
-        if self.udp_server or self.broadcast:
+        if self.udp_server:
+            self.clients.add(new_addr)
+            self.clients_last_alive[new_addr] = time.time()
+        elif self.broadcast:
             self.last_address = new_addr
         return data
 
     def write(self, buf):
         try:
             if self.udp_server:
-                if self.last_address:
-                    self.port.sendto(buf, self.last_address)
+                current_time = time.time()
+                to_remove = set()
+                for address in self.clients:
+                    if self.clients_last_alive[address] + self.timeout > current_time:
+                        self.port.sendto(buf, address)
+                    elif len(self.clients) > 1:
```

Comment:
```suggestion
                    elif len(self.clients) > 1 and len(to_remove) < len(self.clients - 1):
```
If all clients time out at the same time the current behaviour would boot all of them, then once/if one reconnects it would get an endless stream until another one reconnected (assuming the old behaviour of single connections keeping an endless stream is restored, as above). Maybe that's the desired behaviour, but it seems a bit odd. This approach at least keeps the last one, but not sure if such an arbitrary choice is particularly better 🤷